crates/sui-e2e-tests/tests/rest.rs

```diff
@@ -55,3 +64,303 @@ async fn execute_transaction_transfer() {
+
+    let unresolved_transaction = UnresolvedTransaction {
+        ptb: UnresolvedProgrammableTransaction {
+            inputs: vec![UnresolvedInputArgument::Shared {
+                object_id: "0x6".parse().unwrap(),
+                initial_shared_version: None,
+                mutable: None,
+            }],
+            commands: vec![Command::MoveCall(sui_sdk2::types::MoveCall {
+                package: "0x2".parse().unwrap(),
+                module: "clock".parse().unwrap(),
+                function: "timestamp_ms".parse().unwrap(),
+                type_arguments: vec![],
+                arguments: vec![Argument::Input(0)],
+            })],
+        },
+        sender: sender.into(),
+        gas_payment: None,
+        expiration: TransactionExpiration::None,
+    };
+
+    let resolved = client
+        .inner()
+        .resolve_transaction(&unresolved_transaction)
+        .await
+        .unwrap()
+        .into_inner();
+
+    let signed_transaction = test_cluster
+        .wallet
+        .sign_transaction(&resolved.transaction.into());
+    let effects = client
+        .execute_transaction(
+            &ExecuteTransactionQueryParameters::default(),
+            &signed_transaction,
+        )
+        .await
+        .unwrap()
+        .effects;
+
+    assert!(effects.status().is_ok());
+}
+
+#[sim_test]
+async fn resolve_transaction_mutable_shared_object() {
+    let test_cluster = TestClusterBuilder::new().build().await;
+
+    let client = Client::new(test_cluster.rpc_url());
+
+    let (sender, mut gas) = test_cluster.wallet.get_one_account().await.unwrap();
+    gas.sort_by_key(|object_ref| object_ref.0);
+    let obj_to_stake = gas.first().unwrap().0;
+    let validator_address = client
+        .inner()
+        .get_system_state_summary()
+        .await
+        .unwrap()
+        .inner()
+        .active_validators
+        .first()
+        .unwrap()
+        .address;
+
+    let unresolved_transaction = UnresolvedTransaction {
+        ptb: UnresolvedProgrammableTransaction {
+            inputs: vec![
+                UnresolvedInputArgument::Shared {
+                    object_id: "0x5".parse().unwrap(),
+                    initial_shared_version: None,
+                    mutable: None,
+                },
+                UnresolvedInputArgument::ImmutableOrOwned(UnresolvedObjectReference {
+                    object_id: obj_to_stake.into(),
+                    version: None,
+                    digest: None,
+                }),
+                UnresolvedInputArgument::Pure {
+                    value: bcs::to_bytes(&validator_address).unwrap(),
+                },
+            ],
+            commands: vec![Command::MoveCall(sui_sdk2::types::MoveCall {
+                package: "0x3".parse().unwrap(),
+                module: "sui_system".parse().unwrap(),
+                function: "request_add_stake".parse().unwrap(),
+                type_arguments: vec![],
```

Comment:
I think I remember some conversations with @amnn about making type_arguments resolvable.  Not sure if this is always possible, and its not critical for the TS SDK

Reply:
I'm not really sure how to do resolution here. I think that its something that we could possibly add in the future though

Reply:
yeah, I don't think this is critical in the initial version

Reply:
here's an example of splitting a shared coin: https://suiscan.xyz/devnet/tx/54Co9ftVNZyhKQFuPHQ6mNG3GCBw1o7DEGu2P2W7uFWT

Reply:
and one with split and merge: https://suiscan.xyz/devnet/tx/2a1W7uJSkMNc135CH6M5dPc9y6NRmMmVzGVr7VwS8rLt